src/edge/views.py

```diff
@@ -330,6 +330,10 @@ def on_get(self, request, genome_id):
         args = q_parser.parse_args(request)
         field = args['field']
 
+        if not genome.has_location_index:
```

Comment:
What causes the genome index not built when it was first created? random failures?

Reply:
The indices are not required. In fact, because the indices are not shared between genomes, if we build one per fragment all the time, it would remove some of the storage advantage of Edge for engineered genomes. Hence, we don't actually create one when a new genome is created, only when someone go hit the genome on the website, or hit an API that requires it. The issue is that with RO server, if you hit there, you will never end up building the indices. So the solution here is to have building the indices be done via celery, on RO APIs. If a RO server receives an API that's meant for updating, then that's a caller error.